src/sources/source.ts

```diff
@@ -2,13 +2,23 @@
  * Contract of all Sources.
  * Every source must implements the fetch mehod
  * in order to provide the avatar source.
- * 
+ *
  * @export
  * @interface Source
  */
 export interface Source {
     sourceType: string;
     sourceId: string;
+    readonly isAsync: boolean;
     getAvatar(size?: number): string;
 }
 
+/**
+ * Contract of all async sources.
+ * Every async source must implement the processResponse method that extracts the avatar url from the data
+ * @interface AsyncSource
+ */
+export interface AsyncSource extends Source {
```

Comment:
Conflicted, to be honest :)

This way we have to manage the state ourselves, and it's completely possible to have a sync class with `isAsync` set to true. An abstract class could be usefull.

On the other hand, abstract classes in TypeScript are compiled to regular classes in JavaScript, so I tend to avoid them whenever I can, as the only way I can stop the consumer from using them is by documentation.

I'll see if I'm able to set it up in a way that we have an Abstract class, that is not exported outside the module.